python-sdk/tests/test_example_dags.py

```diff
@@ -51,38 +50,34 @@ def session():
     return get_session()
 
 
-DAG_BAG = DagBag(Path(__file__).parent.parent / "example_dags", include_examples=False)
-AIRFLOW_VERSION_INDICATOR = "airflow_version:"
-MINIMUM_AIRFLOW_VERSION = "2.2.5"
+def get_dag_folder() -> Path:
+    """
+    Recursively find the dag folder for the current airflow version.
 
+    :return: the dag folder containing the airflow dags.
+    """
+    base_folder = Path(__file__).parent.parent / "example_dags"
 
-def get_airflow_version(dag: DAG) -> str:
-    for tag in dag.tags:
-        if tag.startswith(AIRFLOW_VERSION_INDICATOR):
-            return tag[len(AIRFLOW_VERSION_INDICATOR) :]
-    return MINIMUM_AIRFLOW_VERSION
+    for path in base_folder.rglob("*"):
+        if path.is_dir():
+            _version = version.parse(path.name)
+            if (
+                isinstance(_version, version.Version)
+                and version.parse(airflow.__version__) >= _version
+            ):
+                return path
 
+    raise ValueError(
+        "Could not find a dag folder with the same name as the current airflow version!"
```

Comment:
Yeah I think with some very minor changes here we could have separate 2.4, 2.3, and 2.2.5 folders.

Reply:
Correct, see https://github.com/astronomer/astro-sdk/pull/1016#discussion_r991217537